src/fetchcode/vcs/__init__.py

```diff
@@ -28,28 +29,34 @@
 
 class VCSResponse:
     """
-    Represent the response from fetching a VCS URL with:
-- `dest_dir`: destination of directory
-- `vcs_type`: VCS Type of URL (git,bzr,hg,svn)
-- `domain` : Source of git VCS (GitHub, Gitlab, Bitbucket)
+        Represent the response from fetching a VCS URL with:
+    - `dest_dir`: destination of directory
+    - `vcs_type`: VCS Type of URL (git,bzr,hg,svn)
+    - `domain` : Source of git VCS (GitHub, Gitlab, Bitbucket)
     """
 
     def __init__(self, dest_dir, vcs_type, domain):
         self.dest_dir = dest_dir
         self.vcs_type = vcs_type
         self.domain = domain
 
+    def delete(self):
+        """
+        Delete the temporary directory.
+        """
+        if os.path.isdir(self.dest_dir):
+            shutil.rmtree(path=self.dest_dir)
+
 
 def fetch_via_vcs(url):
     """
     Take `url` as input and store the content of it at location specified at `location` string
```

Comment:
There is no location arg here? Why create a temp directory and not accept instead a proper location?